modules/standard/Path.chpl

```diff
@@ -610,6 +624,95 @@ proc file.realPath(out error: syserr): string {
   return "";
 }
 
+// Compute the common prefix length between two lists of path components.
+private proc commonPrefixLength(a1: [] string, a2: [] string): int {
+  // Can we use the builtin min/max functions for this?
+  var (min, max) = if a1.size < a2.size then (a1, a2) else (a2, a1);
+  var result = 0;
+
+  for i in 1..min.size do
+    if min(i) != max(i) then
+      return result;
+    else
+      result += 1;
+
+  return result;
+}
+
+/*
+  Returns a relative filepath to `name` either from the current directory or an
+  optional `start` directory. The filesystem is not accessed to verify the
+  existence of the named path or the specified starting location.
+
+  .. warning::
+
+    This method is unsafe for use in a parallel environment due to its
+    reliance on :proc:`locale.cwd()`. Another task on the current locale
+    may change the current working directory at any time.
+
+  :arg name: A path which the caller would like to access.
+  :type name: `string`
+
+  :arg start: The location from which access to name is desired. If no value
+    is provided, defaults to :const:`curDir`.
+  :type start: `string`
+
+  :return: The relative path to `name` from the current directory.
+  :rtype: `string`
+
+  :throws SystemError: Upon failure to get the current working directory.
+*/
+proc relPath(name: string, start:string=curDir): string throws {
+  var realstart = if start == "" then curDir else start;
+
+  // NOTE: Reliance on locale.cwd() can't be avoided.
+  var startComps = absPath(realstart).split(pathSep, -1, true);
+  var nameComps = absPath(name).split(pathSep, -1, true);
```

Comment:
Out of curiosity, have you tested any paths with escaped delimiters, e.g. `/a\/tricky/path` ?

We might encounter issues because we don't yet support something like `shlex.split` (https://github.com/chapel-lang/chapel/issues/11489).

It might be worth adding a test for this if it works, or creating an issue about it if it doesn't.

Reply:
I never even thought of that. I'm glad I asked for more testing.